internal/lexer/lexer.go

```diff
@@ -320,19 +339,35 @@ func lexAssert(nextByte byte, next stateFn) stateFn {
 	}
 }
 
-// lexUntil invokes handler with all bytes up to the stop byte or an eof.
-// The stop byte is not consumed.
-func lexUntil(stop byte, handler func(*Lexer, []byte) stateFn) stateFn {
-	return func(l *Lexer) stateFn {
-		start := l.pos
-		p := bytes.IndexByte(l.input[l.pos:], stop)
-		switch p {
-		case -1:
-			l.pos = l.len
-		default:
-			l.pos += uint32(p)
+// seekUntil Seeks the next stop or eof byte and then return the byte slice between
+// the start position and the end byte.
+func seekUntil(l *Lexer, stop byte) []byte {
```

Comment:
seekUntil and seekDelimited could both use their own unit tests in addition to the coverage offered by TestMetricsLexer

Reply:
Done